toolchain/driver/driver.cpp

```diff
@@ -535,9 +539,22 @@ class Driver::CompilationUnit {
     } else {
       llvm::SmallString<256> output_file_name = options_.output_file_name;
       if (output_file_name.empty()) {
+        if (!source_->is_regular_file()) {
+          // Don't invent file names like `-.o` or `/dev/stdin.o`.
+          driver_->error_stream_
+              << "ERROR: Output file name must be specified for input "
+              << input_file_name_ << " that is not a regular file.\n";
```

Comment:
Seems especially important to quote a bit here given that it may be something weird...

```suggestion
              << "ERROR: Output file name must be specified for input '"
              << input_file_name_ << "' that is not a regular file.\n";
```

Reply:
I think we've used backticks elsewhere, including for filenames. :)

```suggestion
              << "ERROR: Output file name must be specified for input `"
              << input_file_name_ << "` that is not a regular file.\n";
```

Reply:
We use single-quotes a few lines below; going with that for now.